### PR TITLE
[RFR] [developer] Bundle component documentation

### PR DESCRIPTION
**Pull request by eric-chau**
This PR brings Bundle component documentation. The documentation plan:

- What is a bundle?
- Bundle folder structure and entry point
- How the bundle loader works?
  - Default expectations
  - Using recipes to custom some load

### CODE REVIEW

File: doc/developper-documentation/components/bundle.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+# BackBee's bundles
```
Comment (mickaelandrieu) [inline doc/developper-documentation/components/bundle.md]: Can you use # Components instead ? for consistency

File: doc/developper-documentation/components/bundle.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
+# BackBee's bundles
+
+## What is a bundle?
+In BackBee, a bundle is like an BackBee's core application (``BackBee\BBApplication``). It can bring its ``class contents``, ``configurations``, ``listeners``, ``events``, ``templates``, ``controllers``, etc. It's surely even more flexible cause you can use bundle loader recipes to change the way things should be loaded. We will talk about this point later.
```
Comment (mickaelandrieu) [inline doc/developper-documentation/components/bundle.md]: layouts also ? @cedric-b made an issue some days ago .

File: doc/developper-documentation/components/bundle.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# BackBee's bundles
+
+## What is a bundle?
+In BackBee, a bundle is like an BackBee's core application (``BackBee\BBApplication``). It can bring its ``class contents``, ``configurations``, ``listeners``, ``events``, ``templates``, ``controllers``, etc. It's surely even more flexible cause you can use bundle loader recipes to change the way things should be loaded. We will talk about this point later.
+While the core application contains many features to build up your website/application, a bundle __must__ contain only one main functionnality. For example, a ArticleDisqusBundle that brings article content and Disqus commenting feature __must not__ exist. A good solution is to split it into two distincts bundles:
+
+- ArticleBundle: could be a bundle that brings article classcontent and its views, maybe also some listeners to alter some behaviors.
+- DisqusBundle: could simply be an implementation of Disqus API into BackBee.
+
+Also, if one day you need the Disqus feature in another project that doesn't need the article class content, this will result on embedding unused code. So keep it in mind: __one bundle = one feature__.
```
Comment (mickaelandrieu) [inline doc/developper-documentation/components/bundle.md]: Agree.

File: doc/developper-documentation/components/bundle.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+```
+
+So we saw what's a bundle and what is needed to create one. Let's learn how register it to BackBee. You need to create ``bundles.yml`` into your project config folder (generally located at ``PROJECT_ROOT_DIR/repository/Config/``):
+
+```yaml
+# PROJECT_ROOT_DIR/repository/Config/bundles.yml
+nl: BackBee\Bundle\NewsletterBundle\Newsletter
+demo: BackBee\Bundle\DemoBundle\Demo
+```
+That's all! Pretty simple, right? One more thing, bundle are automatically loaded into application service container. Your bundle identifier will always follow this pattern: ``bundle.__BUNDLE_NAME__``. Note that the ``__BUNDLE_NAME__`` is the declared key of your bundle, so it's ``bundle.nl`` in this case (``nl`` cause it's the key name of the bundle entry point in ``bundles.yml``). You can also access to NewsletterBundle's config from the service container. Its service identifier follow this pattern: ``bundle.__BUNDLE_NAME__.config``. So the NewsletterBundle config identifier is ``bundle.nl.config``.
```
Comment (mickaelandrieu) [inline doc/developper-documentation/components/bundle.md]: bundles

File: doc/developper-documentation/components/bundle.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+The first task the bundle loader do is to get every declared bundle containing by ``PROJECT_ROOT_DIR/repository/Config/bundles.yml`` and load them one by one. Note that the order bundles are loaded is determined by declaration order in ``bundles.yml`` (in previous example, NewsletterBundle will be loaded first). The order is important cause the last loaded one will always win if there is a conflict about which template to use or which parameter value to use in service container. So __be careful about your bundle declaration order__.
+
+From now, consider that the current path (``./``) is ``__BUNDLE_BASE_DIR__``. So ``./Config/`` is equal to ``__BUNDLE_BASE_DIR__/Config/``.
+
+#### Default expectations
+We are now coming up to talk about bundle default structure expected by the loader:
+
+- services for dependency injection container can be declared in yaml or xml format and is expected to be find into ``./Config/services.[yml|xml]``
```
Comment (mickaelandrieu) [inline doc/developper-documentation/components/bundle.md]: again, can we provide Layouts in Bundles ?

Comment (eric-chau) [inline doc/developper-documentation/components/bundle.md]: I didn't mention load of layouts with bundles at moment cause it's not possible yet, the issue still not resolved.

File: doc/developper-documentation/components/bundle.md (hunk 4 above)
Comment (eric-chau) [inline doc/developper-documentation/components/bundle.md]: nice catch

**Comment by eric-chau** [comment]
Thank you @mickaelandrieu for your review, I updated it.

**Comment by mickaelandrieu** [comment]
:+1: *but* you need to register your page into the Component section (did you read the README ?)

**Comment by eric-chau** [comment]
Done, sorry it's my bad yeah.

**Comment by mickaelandrieu** [comment]
thank you @eric-chau

**Pull request merged**